flake.nix

```diff
@@ -0,0 +1,178 @@
+          (ghcPkgs.callCabal2nix "clash-cosim" ./clash-cosim {
+            inherit (self.packages.${system}) clash-prelude;
+          }).overrideAttrs(old: {
+            # Ensure iverilog is in the path, since Setup.hs checks for vvp.
+            nativeBuildInputs = (old.nativeBuildInputs or []) ++ [
+              pkgs.verilog
+            ];
+          });
+
+        clash-ffi =
+          ghcPkgs.callCabal2nix "clash-ffi" ./clash-ffi {
+            inherit (self.packages.${system}) clash-prelude;
+          };
+
+        clash-ghc =
+          ghcPkgs.callCabal2nix "clash-ghc" ./clash-ghc {
+            inherit (self.packages.${system}) clash-prelude clash-lib;
+          };
+
+        clash-lib =
+          ghcPkgs.callCabal2nix "clash-lib" ./clash-lib {
+            inherit (self.packages.${system}) clash-prelude;
+          };
+
+        clash-lib-hedgehog =
+          ghcPkgs.callCabal2nix "clash-lib-hedgehog" ./clash-lib-hedgehog {
+            inherit (self.packages.${system}) clash-lib;
+          };
+
+        clash-prelude =
+          ghcPkgs.callCabal2nix "clash-prelude" ./clash-prelude {};
+
+        clash-prelude-hedgehog =
+          ghcPkgs.callCabal2nix "clash-prelude-hedgehog" ./clash-prelude-hedgehog {
+            inherit (self.packages.${system}) clash-prelude;
+          };
+
+        clash-term =
+          ghcPkgs.callCabal2nix "clash-term" ./clash-term {
+            inherit (self.packages.${system}) clash-lib;
+          };
+
+        clash-testsuite =
+          ghcPkgs.callCabal2nix "clash-testsuite" ./tests {
+            inherit (self.packages.${system})
+              clash-cores
+              clash-ghc
+              clash-lib
+              clash-prelude;
+          };
+
+        default = self.packages.${system}.clash-ghc;
+      };
+
+      apps = {
+        clash = {
+          type = "app";
+          program = "${self.packages.${system}.clash-ghc}/bin/clash";
+        };
+
+        clashi = {
+          type = "app";
+          program = "${self.packages.${system}.clash-ghc}/bin/clashi";
+        };
+
+        clash-term = {
+          type = "app";
+          program = "${self.packages.${system}.clash-term}/bin/clash-term";
+        };
+
+        clash-testsuite = {
+          type = "app";
+          program = "${self.packages.${system}.clash-testsuite}/bin/clash-testsuite";
+        };
+
+        default = self.apps.${system}.clashi;
+      };
+
+      devShells = {
+        default =
+          pkgs.mkShell {
+            inputsFrom = [
+              self.packages.${system}.clash-cores.env
+              self.packages.${system}.clash-cosim.env
+              self.packages.${system}.clash-ffi.env
+              self.packages.${system}.clash-ghc.env
+              self.packages.${system}.clash-lib.env
+              self.packages.${system}.clash-lib-hedgehog.env
+              self.packages.${system}.clash-prelude.env
+              self.packages.${system}.clash-prelude-hedgehog.env
```

Comment:
Will this build any of the packages when invoking a `nix-shell`? Given that, e.g., one of `clash-lib`'s dependencies is `clash-prelude`.

Reply:
No, it doesn't build any of these. It just ensures the started shell has an environment capable of building all these packages. You could deduplicate a little by removing say, `clash-lib` and `clash-prelude` since they depended on by `clash-ghc`, or `clash-prelude-hedgehog` since it's depended on by `clash-cores`, but I think it's easier reading to just say _here are all the packages we might want to build, you get everything needed to build them in your shell_